File: chapitre-11/exemple-signal.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-signal.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------

from __future__ import print_function
import signal
import os

def handler(num, stack):
	print("Signal", num, "received")

for sig in range(1, signal.NSIG):
	try:
		signal.signal(sig, handler)
	except:
		print("Unable to catch signal", sig)

while True:
	signal.pause()

File: chapitre-11/exemple-signal.c
// ------------------------------------------------------------------
// exemple-signal.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <unistd.h>

void signal_handler(int num)
{
	fprintf(stdout, "\n%ld received signal: %d (%s)\n", 
	        (long) getpid(), num, sys_siglist[num]);
}

int main (void)
{
	int i;

	for (i = 1; i < NSIG; i ++)
		if (signal(i, signal_handler) == SIG_ERR)
			fprintf(stderr, "Signal %d not handled\n", i);

	while (1)
		pause();
}

